### ISSUE TITLE
ui-route problem

### ISSUE THREAD

**Issue by adyba** [delta98/ionic-material-design-lite#70]
Hi, the theme looks promising but it somehow cripples the Ionic bundled ui-router.
My `<head>` (1st September ionic-material-design-lite#0.5.3) on brand new Ionic tab template looks like this

```
<head>
    <meta charset="utf-8">
    <meta name="viewport" content="initial-scale=1, maximum-scale=1, user-scalable=no, width=device-width">
    <title></title>
    <link rel="stylesheet" href="http://fonts.googleapis.com/css?family=Roboto:300,400,500,700" rel="stylesheet" type="text/css">
    <link href="lib/ionic/release/css/ionic.css" rel="stylesheet">
    <link href="lib/ionic-material-design-lite/dist/css/ionic.material-design-lite.min.css" rel="stylesheet">
    <link href="css/style.css" rel="stylesheet">

    <!-- IF using Sass (run gulp sass first), then uncomment below and remove the CSS includes above
    <link href="css/ionic.app.css" rel="stylesheet">
    -->

    <!-- ionic/angularjs js -->
    <script src="lib/ionic/release/js/ionic.bundle.js"></script>
    <script src="lib/ionic-material-design-lite/dist/js/ionic.material-design-lite.min.js"></script>
    <!-- cordova script (this will be a 404 during development) -->
    <script src="cordova.js"></script>

    <!-- your app's js -->
    <script src="js/app.js"></script>
    <script src="js/controllers.js"></script>
    <script src="js/services.js"></script>
  </head>
```
The application handles just one complete tab change. Then after it changes only the tab (active state of the tab + header line) but the view template remains the same. Once I comment out all the `ionic.material-design-lite` stuff, everything works as expected. Unfortunately the console is empty, there is no error message, which makes it difficult to trace the bug.

Would anybody be so nice and publish a working (ie. tab) starter as a plunk? The codepen sample provided uses a some mysterious pieces of javascript that makes it (the pen) work but it is difficult to replicate it in a real Ionic app.

**Comment by delta98** [comment]
Can you provide your ui-router setup because it MDL doesn't have any interaction with ui-sref or events

**Comment by adyba** [comment]
Sure, its a mint Ionic app generated by Tab template, so there is no mystery.
 
```
.config(function($stateProvider, $urlRouterProvider) {
$stateProvider

    .state('tab', {
    url: '/tab',
    abstract: true,
    templateUrl: 'templates/tabs.html'
  })

  .state('tab.dash', {
    url: '/dash',
    views: {
      'tab-dash': {
        templateUrl: 'templates/tab-dash.html',
        controller: 'DashCtrl'
      }
    }
  })

  .state('tab.chats', {
      url: '/chats',
      views: {
        'tab-chats': {
          templateUrl: 'templates/tab-chats.html',
          controller: 'ChatsCtrl'
        }
      }
    })
    .state('tab.chat-detail', {
      url: '/chats/:chatId',
      views: {
        'tab-chats': {
          templateUrl: 'templates/chat-detail.html',
          controller: 'ChatDetailCtrl'
        }
      }
    })

  .state('tab.account', {
    url: '/account',
    views: {
      'tab-account': {
        templateUrl: 'templates/tab-account.html',
        controller: 'AccountCtrl'
      }
    }
  });

  $urlRouterProvider.otherwise('/tab/dash');
});
```

**Comment by acamenhas** [comment]
I've replicated your experiment and i also checked that that happens.
Did you notice also that the styling does not appear?

**Comment by delta98** [comment]
I will take a look at this tonight as I've started to build default apps taken from the ionic base templates and adding the required scripts.

I've not actually noticed an issue with tab routing though.

**Comment by delta98** [comment]
Taken a look at a straight forward install from the ionic cli (http://ionicframework.com/docs/cli/start.html) the state sticks on tab state change. It's not related to the library.

**Comment by adyba** [comment]
Jason, would you please share your successful experiment with us (via Plunker, Git or whatever)? This should be something minor like order of libraries in `<head>` or something. Unfortunately as the code is fresh and not yet "talkative" to the console its hard to track the issue. Do you want me to create the sample/repository? We just need something real to discuss, sending statements back and forth it is workin` not workin` wont help anybody.
There must be something as even the ripple effect and other JS features are not showing  @acamenhas .

**Comment by delta98** [comment]
This is a known issue with ionic caching and *not* this project. Please refer to http://stackoverflow.com/questions/28827928/template-does-not-update-when-using-ui-router-and-ion-tabs

**Comment by adyba** [comment]
Nope, the problem was my lost sync during the release period. I have started my trials on August 31st several hours before the release with separated MDL files. On September 1st I updated just your project, deleted the MDL, kept the JS file reference which fired half of the code but obviously not the bundled MDL. Started from scratch, following the updated `README.md` instructions it works just fine. The SO you mentioned is not relevant with current Ionic version that is installed by (ionic CLI) default.

As a side effect I have created a Ionic Tab template clone on Plunker. The Public branch is here http://plnkr.co/edit/P2nauG.
